### PR TITLE
YES! THIS IS WHAT I'M SUPPOSED TO PUSH!

### PR DESCRIPTION
**Pull request by Guy-Markman**
please review

### CODE REVIEW

**Comment by Guy-Markman** [comment]
@alonbl 
This is what I did until now. Can you review it?

File: 01-zombie.py [hunk 1]
```diff
@@ -15,16 +15,19 @@ def child_process():
 
 
 def parent_process(child_pid):
-    util.write_to_target(1, "Parent pid: %s\nChild pid: %s\n" % (os.getpid(), child_pid))
-    time.sleep(TIME_SLEEP_PARENT)    
+    util.write_to_target(
+        1,
+        "Parent pid: %s\nChild pid: %s\n" % (os.getpid(), child_pid),
```
Comment (alonbl) [inline 01-zombie.py]: better format is:
```
    (
        "Parent pid: %d\n"
        "Child pid: %d\n"
    ) % (
        os.getpid(),
        child_pid
    )

File: 04-eintr.py [hunk 2]
```diff
@@ -20,14 +21,12 @@ def parent_process(child_pid):
     util.write_to_target(
         1, "Parent pid: %s\nChild pid: %s\n" % (os.getpid(), child_pid)
         )
-    time.sleep(TIME_SLEEP_PARENT)
+    util.read_from_target(0, BUFF_SIZE)
```
Comment (alonbl) [inline 04-eintr.py]: do you actually read stdin from both parent and child?

File: 04-eintr.py [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 import os
-import time
 import sys
 import util
 import signal
 
 TIME_SLEEP_CHILD = 2
 TIME_SLEEP_PARENT = 60
 
+BUFF_SIZE = 1234
```
Comment (alonbl) [inline 04-eintr.py]: strange number

File: 08-daemon.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python
+import signals
+import os
+
+
+def main():
```
Comment (alonbl) [inline 08-daemon.py]: ?

File: util.py [hunk 5]
```diff
@@ -1,7 +1,18 @@
...
         text = text[os.write(target, text):]
+
+
+def read_from_target(target, size):
+    data = None
+    while data is not None:
+        try:
+            data = os.read(target, size)
+        except IOError as e:
+            if e.errno != errno.EINTR:
```
Comment (alonbl) [inline util.py]: yes, this is correct

File: 08-daemon.py (hunk 4 above)
Comment (Guy-Markman) [inline 08-daemon.py]: I'll work on it

File: 01-zombie.py (hunk 1 above)
Comment (Guy-Markman) [inline 01-zombie.py]: Changed

File: 04-eintr.py (hunk 3 above)
Comment (Guy-Markman) [inline 04-eintr.py]: Changed

File: 04-eintr.py (hunk 2 above)
Comment (Guy-Markman) [inline 04-eintr.py]: Changed

**Comment by Guy-Markman** [comment]
@alonbl I got conflicts. How do I fix it? I only mess up my repo when I tried to fix it myself

**Comment by alonbl** [comment]
the configs are there because you probably did not start with fresh repo
at the end you will be able to push with --force to override whatever in repo

File: 08-daemon.py [hunk 6]
```diff
@@ -1,9 +1,28 @@
 #!/usr/bin/env python
-import signals
 import os
+import sys
+import resource
+
+def set_up():
+    os.closerange(3, resource.RLIMIT_NOFILE+1)  # +1 for execlude
```
Comment (alonbl) [inline 08-daemon.py]: you do not need the +1 as the nofile is the max
closerange is not system call, but...

File: 08-daemon.py [hunk 7]
```diff
@@ -1,9 +1,28 @@
...
+    new_directory = os.open("/dev/null", os.O_RDWR)
+    for x in range (3):
+        os.dup2(x, new_directory)
```
Comment (alonbl) [inline 08-daemon.py]: and? what is missing?

Comment (Guy-Markman) [inline 08-daemon.py]: Many more is missing. I'm still working on it. It's just the starting version. I'll upload the complete version into Alon-Review.

**Pull request merged**